research-software-directory/RSD-as-a-service
# rsd admin can edit package manager item

# Rsd admin can edit package manager item

Closes #1277

Changes proposed in this pull request:
* Rsd admin can edit package manager item. It can: 
    * disable service by adding a comment into designated input field. The input is enabled only for the services that are "supported". **The info about "supported services" is kept locally (in an data object) and need to be manually updated.**
    * change "detected platform" using a drop-down options. The list of supported providers is kept locally (in same data object) and need to be manually updated. 
* When updating package manager entry only platform and the reasons for disabling are updated

How to test:
* `make start` to build app and create test data
* login as rsd_admin, first user should become rsd_admin automatically if you have correct .env settings
* open first software item and navigate to package managers section
* confirm you can edit existing item
* confirm you can provide reason to stop scraping
* confirm that disabling is visible on the PackMan page
* confirm that disabling is visible on the background services page
* confirm you can create new item
* login as regular rsd user
* edit or create software item
* navigate to package manager section
* confirm you cannot edit package manager item
* confirm you can delete and add new items

## Add package manager (not changed)
![image](https://github.com/user-attachments/assets/ce8b3544-bbce-48ba-b216-716d8c321769)

## Admin package manager modal (to disable service or change platform)
![image](https://github.com/user-attachments/assets/dc17d7fc-245a-443b-b7ad-0cfb395d2b22)

## Example disabled service on PM page
![image](https://github.com/user-attachments/assets/96900cd4-f269-4db2-8b9f-ad6deeeea1af)

## Example disabled service on services page
![image](https://github.com/user-attachments/assets/d631f8fc-d9c6-43a0-8476-cc791fc37876)

PR Checklist:
